File: 02/Java/Rectangle.java
public class Rectangle extends Tetragon {

    public Rectangle(double side1, double side2) {
        super(side1, side2, side1, side2, Math.PI / 2);
    }

    @Override
    public surfaceArea() {
        return sideLengths[0] * sideLengths[1];
    }

    @Override
    public perimeter() {
        return 2 * sideLengths[0] + 2 * sideLengths[1];
    }

}

File: 02/cpp/Rectangle.cpp
#include "Rectangle.hpp"

#include <cmath>

static const double pi = std::acos(-1);

Rectangle::Rectangle(double side1, double side2) :
    Tetragon(side1, side2, side1, side2, pi / 2) {}

double Rectangle::surface_area() noexcept { return sideLengths[0] * sideLengths[1]; }

double Rectangle::perimeter() noexcept { return 2 * sideLengths[0] + 2 * sideLengths[1]; }